x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/time_scaling.tsx

```diff
@@ -0,0 +1,187 @@
+  EuiPopover,
+  EuiButtonEmpty,
+  EuiSpacer,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useState } from 'react';
+import { StateSetter } from '../../types';
+import { IndexPatternColumn, operationDefinitionMap } from '../operations';
+import { mergeLayer } from '../state_helpers';
+import { unitSuffixesLong } from '../suffix_formatter';
+import { TimeScaleUnit } from '../time_scale';
+import { IndexPatternPrivateState } from '../types';
+
+const DEFAULT_TIME_SCALE = 'm' as const;
+
+export function TimeScaling({
+  selectedColumn,
+  columnId,
+  layerId,
+  state,
+  setState,
+}: {
+  selectedColumn: IndexPatternColumn;
+  columnId: string;
+  layerId: string;
+  state: IndexPatternPrivateState;
+  setState: StateSetter<IndexPatternPrivateState>;
+}) {
+  const [popoverOpen, setPopoverOpen] = useState(false);
+  const layer = state.layers[layerId];
+  const hasDateHistogram = layer.columnOrder.some(
+    (colId) => layer.columns[colId].operationType === 'date_histogram'
+  );
+  const selectedOperation = operationDefinitionMap[selectedColumn.operationType];
+  if (
+    !selectedOperation.timeScalingMode ||
+    selectedOperation.timeScalingMode === 'disabled' ||
+    !hasDateHistogram
+  ) {
+    return null;
+  }
+
+  if (!selectedColumn.timeScale) {
+    return (
+      <EuiText textAlign="right">
+        <EuiSpacer size="s" />
+        <EuiPopover
+          ownFocus
+          button={
+            <EuiButtonEmpty
+              size="xs"
+              iconType="arrowDown"
+              iconSide="right"
+              data-test-subj="indexPattern-time-scaling-popover"
+              onClick={() => {
+                setPopoverOpen(true);
+              }}
+            >
+              {i18n.translate('xpack.lens.indexPattern.timeScale.advancedSettings', {
+                defaultMessage: 'Add advanced options',
+              })}
+            </EuiButtonEmpty>
+          }
+          isOpen={popoverOpen}
+          closePopover={() => {
+            setPopoverOpen(false);
+          }}
+        >
+          <EuiText size="s">
+            <EuiLink
+              data-test-subj="indexPattern-time-scaling-enable"
+              color="text"
+              onClick={() => {
+                setState(
+                  mergeLayer({
+                    state,
+                    layerId,
+                    newLayer: {
+                      ...state.layers[layerId],
+                      columns: {
+                        ...state.layers[layerId].columns,
+                        [columnId]: {
+                          ...selectedColumn,
+                          timeScale: DEFAULT_TIME_SCALE,
+                        },
+                      },
+                    },
+                  })
+                );
+              }}
+            >
+              {i18n.translate('xpack.lens.indexPattern.timeScale.enableTimeScale', {
+                defaultMessage: 'Show as rate',
+              })}
```

Comment:
The text here uses inconsistent words, going from `show as rate` to `normalize by unit`. My preference is to call these "time units" in both places, which would make this text change:

'Show as rate' becomes 'Show with normalized time units'
'Normalize by unit' becomes 'Normalize by time units'